pkg/prometheus/promcfg.go

```diff
@@ -623,6 +630,41 @@ func (cg *ConfigGenerator) Generate(
 	return yaml.Marshal(cfg)
 }
 
+func (cg *ConfigGenerator) appendStorageSettingsConfig(cfg yaml.MapSlice, p *v1.Prometheus) (yaml.MapSlice, error) {
+	isExemplarsEnabled := false
+	for _, featureFlag := range p.Spec.EnableFeatures {
+		if featureFlag == "exemplar-storage" {
+			isExemplarsEnabled = true
+			break
+		}
+	}
+
+	if !isExemplarsEnabled {
+		return cfg, nil
```

Comment:
Since Prometheus starts without problem if the exemplars config is defined and without the `exemplar-storage` flag, the operator could also not check anything.

Reply:
perfect, I will apply your comments and fix the build.

Reply:
done :)